cmake/thirdparty/SetupHIP.cmake

```diff
@@ -15,24 +15,77 @@ find_package(HIP REQUIRED)
+elseif(${HIP_PLATFORM} STREQUAL "nvcc")
+    set(HIP_RUNTIME_DEFINES "-D__HIP_PLATFORM_NVCC__")
+    find_package(cudatoolkit)
+    set(HIP_RUNTIME_LIBRARIES "${CUDAToolkit_LIBRARY_DIR}/libcudart.so")
+    set(HIP_RUNTIME_INCLUDE_DIRS "${HIP_RUNTIME_INCLUDE_DIRS};${CUDAToolkit_INCLUDE_DIR}")
 endif()
 if ( IS_DIRECTORY "${HIP_ROOT_DIR}/hcc/include" ) # this path only exists on older rocm installs
-        set(HIP_RUNTIME_INCLUDE_DIRS "${HIP_ROOT_DIR}/include;${HIP_ROOT_DIR}/hcc/include" CACHE STRING "")
+    set(HIP_RUNTIME_INCLUDE_DIRS "${HIP_ROOT_DIR}/include;${HIP_ROOT_DIR}/hcc/include" CACHE STRING "")
 else()
-        set(HIP_RUNTIME_INCLUDE_DIRS "${HIP_ROOT_DIR}/include" CACHE STRING "")
+    set(HIP_RUNTIME_INCLUDE_DIRS "${HIP_ROOT_DIR}/include" CACHE STRING "")
+endif()
+set(HIP_RUNTIME_COMPILE_FLAGS "${HIP_RUNTIME_COMPILE_FLAGS};-Wno-unused-parameter")
+
+set(_hip_compile_flags " ")
+if (ENABLE_CLANG_HIP)
+    if (NOT (${HIP_PLATFORM} STREQUAL "clang"))
+        message(FATAL_ERROR "ENABLE_CLANG_HIP requires HIP_PLATFORM=clang")
+    endif()
+    set(_hip_compile_flags -x;hip)
+    # Using clang HIP, we need to construct a few CPP defines and compiler flags
+    foreach(arch ${BLT_CLANG_HIP_ARCH})
+        string(TOUPPER ${arch} UPARCH)
+        string(TOLOWER ${arch} lowarch)
+        list(APPEND _hip_compile_flags "--offload-arch=${lowarch}")
+        set(_hip_compile_defines "${HIP_RUNTIME_DEFINES};-D__HIP_ARCH_${UPARCH}__=1")
+    endforeach(arch)
+    
+    # We need to pass rocm path as well, for certain bitcode libraries.
+    # First see if we were given it, then see if it exists in the environment.
+    # If not, don't try to guess but print a warning and hope the compiler knows where it is.
+    if (NOT ROCM_PATH)
+        find_path(ROCM_PATH
+            bin/rocminfo
+            ENV ROCM_DIR
+            ENV ROCM_PATH
+            ${HIP_ROOT_DIR}/../
+            ${ROCM_ROOT_DIR}
+            /opt/rocm)
+    endif()
+
+    if(DEFINED ROCM_PATH)
+        list(APPEND _hip_compile_flags "--rocm-path=${ROCM_PATH}")
+    else()
+        message(WARN "ROCM_PATH not set or found! This is typically required for Clang HIP Compilation")
+    endif()
+
+    message(STATUS "Clang HIP Enabled. Clang flags for HIP compilation: ${_hip_compile_flags}")
+    message(STATUS "Defines for HIP compilation: ${_hip_compile_defines}")
+
+    blt_import_library(NAME             hip
+                       DEFINES          ${_hip_compile_defines}
+                       COMPILE_FLAGS    ${_hip_compile_flags}
+                       DEPENDS_ON       ${HIP_RUNTIME_LIBRARIES})
+else()
+
+    # depend on 'hip', if you need to use hip
+    # headers, link to hip libs, and need to run your source
+    # through a hip compiler (hipcc)
+    blt_register_library(NAME      hip
+                        INCLUDES  ${HIP_INCLUDE_DIRS}
+                        LIBRARIES ${HIP_LIBRARIES}
+                        TREAT_INCLUDES_AS_SYSTEM ON)
```

Comment:
This over-complication of the non-clang path seems to cause the regular hipcc breakage. I'll need to fix and update.